cogs/tmpc.py

```diff
@@ -230,15 +233,18 @@ async def un_lock_helper(overwrites, document, guild, key, connect):
                   brief="Renames the Channel",
                   help="Renames the channel",
                   aliases=["rn", "mv"])
-    @cooldown(1, 60, BucketType.user)
+    @cooldown(1, 60, BucketType.channel)
     async def rename(self, ctx: Context, *, name: str):
         """
         Renames a study/gaming channel
         """
-        document = await self.check_tmpc_channel(ctx, is_mod=True)
+        document = await self.check_tmpc_channel(ctx.author, ctx.channel.id, is_mod=True)
+        names = {channel.name.lower() for channel in ctx.guild.channels}
```

Comment:
maybe check only the channels that are in the same category? Or do we really need to check the whole server?

Reply:
Ok yeah fine